chore: remove arg and env variable combination

jacderida opened the pull request:
Our code was setup such that the `--peer` arguments and those defined using `SAFE_PEERS` were combined together to form one list. Now we use `clap` to parse one or the other.

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 15 Jan 24 21:47 UTC
This pull request removes the combination of `--peer` arguments and `SAFE_PEERS` environment variables to form a single peer list. It now uses `clap` to parse either one or the other.
<!-- reviewpad:summarize:end -->

davidrusu commented on sn_client/src/lib.rs:
```diff
@@ -156,14 +156,11 @@ where
 // TODO: dedupe this w/ sn_node?
-#[cfg(any(test, feature = "test-utils"))]
 /// Initialise logger for tests, this is run only once, even if called multiple times.
 pub fn init_test_logger() {
```
Is this used in non-test code? maybe move it into the `mod tests { ..}` block?

RolandSherwin replied:
It is used in examples, so cannot have it inside `cfg(test)`. Also there seems to be 4 definitions of `init_test_logger` in the code, maybe shall I have a single one inside `sn_interface` under `feature="test"` and import it in other places?

joshuef replied:
> maybe shall I have a single one inside sn_interface under feature="test" and import it in other places?

That sounds good. But then it's probably just `"init_logger"` perhaps?

joshuef commented:
@RolandSherwin why do we need `cfg(feature="test")` vs something like `cfg(feature="utils")`? 

What are those functions doing? (I just want to be sure they are actually testing code... ie faking some situation, vs just utility code)

joshuef commented:
> Ah! So for example, functions like init_test_logger() should be under utils while fns like create_test_client_with() should be under test?

As per a HO just now, let's leave off renaming/classifying exports and keep this PR tighter to just `run crate tests with no feature flags` as the aim.

Then later on we can check that everything we're exporting makes sense and classify them a bit better 👍

jacderida committed:
chore: remove arg and env variable combination

Our code was setup such that the `--peer` arguments and those defined using `SAFE_PEERS` were
combined together to form one list. Now we use `clap` to parse one or the other.
